src/panels/RemoveSecretPanel/Fields/Fields.tsx

```diff
@@ -0,0 +1,94 @@
+import { Select, ConfirmationModal } from "@canonical/react-components";
+import { useFormikContext } from "formik";
+import { useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
+import usePortal from "react-useportal";
+
+import FormikField from "components/FormikField/FormikField";
+import type { EntityDetailsRoute } from "components/Routes/Routes";
+import { getSecretByURI, getModelUUIDFromList } from "store/juju/selectors";
+import { useAppSelector } from "store/store";
+
+import type { FormFields } from "../types";
+
+export enum Label {
+  CANCEL_BUTTON = "Cancel",
+  CONFIRM_BUTTON = "Remove",
+  CONFIRM_TITLE = "Remove secret?",
+  REMOVE_ALL = "Remove all revisions",
+  REVISION = "Revision",
+}
+
+type Props = {
+  hideConfirm: () => void;
+  handleRemoveSecret: (values: FormFields) => void;
+  secretURI?: string | null;
+  showConfirm: boolean;
+};
+
+const Fields = ({
+  hideConfirm,
+  handleRemoveSecret,
+  secretURI,
+  showConfirm,
+}: Props): JSX.Element => {
+  const { userName, modelName } = useParams<EntityDetailsRoute>();
+  const modelUUID = useSelector(getModelUUIDFromList(modelName, userName));
+  const { values } = useFormikContext<FormFields>();
+  const secret = useAppSelector((state) =>
+    getSecretByURI(state, modelUUID, secretURI),
+  );
+  const { Portal } = usePortal();
+
+  return (
+    <>
+      <FormikField
+        id={Label.REMOVE_ALL}
+        label={Label.REMOVE_ALL}
+        name="removeAll"
+        type="checkbox"
+        disabled={secret?.revisions.length === 1}
```

Comment:
I was thinking it could be confusing if the checkbox was enabled as it's giving you a choice and the user might wonder if there will be a different outcome depending on which option they choose. I wonder if there is only one revision we should hide both fields and display some text instead? Maybe something like: "This secret has one revision (7) and will be completely removed."

Reply:
That's a good idea! If it is not that lengthy to implement, I think it might improve the user experience if done.

Reply:
Done.